numpy/ctypeslib.py

```diff
@@ -401,8 +401,9 @@ def prep_pointer(pointer_obj, shape):
         attach an __array_interface__ property to it if it does not
         yet have one.
         """
-        try: pointer_obj.__array_interface__
+        try: pointer_obj.__array_interface__['shape'] = shape
         except AttributeError: pass
+        except TypeError: return
```

Comment:
Is `__array_interface__ ` allowed to not be a dictionary? Seems to me that we should not silence that error